assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -10,28 +10,108 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
+import { useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Draggable from './draggable';
+import { isBlockAllowedOnPage, useMoveBlockToPage } from '../../helpers';
+
+const BlockDraggable = ( { children, clientId, blockName, blockElementId, onDragStart, onDragEnd } ) => {
+	const { rootClientId } = useSelect( ( select ) => select( 'core/block-editor' ).getBlockRootClientId( clientId ) );
+	const { index } = useSelect( ( select ) => select( 'core/block-editor' ).getBlockIndex( clientId, rootClientId ) );
 
-const BlockDraggable = ( { children, clientId, blockName, rootClientId, blockElementId, index, onDragStart, onDragEnd } ) => {
 	const transferData = {
 		type: 'block',
 		srcIndex: index,
 		srcRootClientId: rootClientId,
 		srcClientId: clientId,
 	};
 
+	const { moveBlockToPage, getPageByOffset } = useMoveBlockToPage( clientId );
+
+	// This holds the currently highlighted element, if any
+	const hoverElement = useRef( { pageId: null, element: null, classes: [] } );
+
+	/**
+	 * Clear highlighting of pages.
+	 */
+	const clearHighlight = () => {
+		// Unhighlight old highlighted page.
+		if ( hoverElement.current.element ) {
+			hoverElement.current.element.classList.remove( ...hoverElement.current.classes );
+		}
+
+		// Reset current hover element
+		hoverElement.current = { pageId: null, element: null, classes: [] };
+	};
+
+	/**
+	 * On the page given by the offset, set classes indicating that an element drag on this page is in progress, which element is being dragged and whether a drop is allowed on this page.
+	 *
+	 * @param {number} offset  Integer specifying offset from current page - e.g. -2 on page 3 will return id of page 1. Page indices are zero-based.
+	 */
+	const setHighlightByOffset = ( offset ) => {
+		// Get neighboring page.
+		const pageId = getPageByOffset( offset );
+		const hasHighlightChanged = pageId !== hoverElement.current.pageId;
+
+		if ( ! hasHighlightChanged ) {
+			return;
+		}
+
+		clearHighlight();
+
+		// Highlight page and mark whether drop is allowed or not.
+		if ( pageId ) {
+			// Drop is always allowed on the initial page (offset=0)
+			const isAllowed = offset === 0 || isBlockAllowedOnPage( blockName, pageId );
+			const classes = [
```

Comment:
Should this use / import the `classnames` function?

Reply:
I did that initially, but as I need an array and not a merged string (`classList.add/.remove` doesn't accept `("foo bar")`, only `("foo", "bar")`), I would need to split it anyway - and then `classNames` was doing nothing essentially.